### PR TITLE
Move cli to arcade

### PR DESCRIPTION
**Pull request by livarcocc**
This is a huge change, sorry for that. For those reviewing it, I would recommend reviewing by commit. You can follow how I went about getting things working here.

### CODE REVIEW

**Comment by livarcocc** [comment]
@dotnet/dotnet-cli This is now good to be reviewed.

File: src/dotnet/dotnet.csproj [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <TargetFramework>$(CliTargetFramework)</TargetFramework>
     <AssemblyName>dotnet</AssemblyName>
     <OutputType>Exe</OutputType>
-    <AssemblyOriginatorKeyFile>../../tools/Key.snk</AssemblyOriginatorKeyFile>
+    <StrongNameKeyId>MicrosoftAspNet</StrongNameKeyId>
```
Comment (nguerrera) [inline src/dotnet/dotnet.csproj]: >MicrosoftAspNet [](start = 21, length = 15)

Is this the same key as before?

File: src/Microsoft.DotNet.Cli.Sln.Internal/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -4,7 +4,7 @@
 using System.Reflection;
 using System.Runtime.CompilerServices;
 
-[assembly: InternalsVisibleTo("Microsoft.DotNet.Cli.Sln.Internal.Tests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100039ac461fa5c82c7dd2557400c4fd4e9dcdf7ac47e3d572548c04cd4673e004916610f4ea5cbf86f2b1ca1cb824f2a7b3976afecfcf4eb72d9a899aa6786effa10c30399e6580ed848231fec48374e41b3acf8811931343fc2f73acf72dae745adbcb7063cc4b50550618383202875223fc75401351cd89c44bf9b50e7fa3796")]
+[assembly: InternalsVisibleTo("Microsoft.DotNet.Cli.Sln.Internal.Tests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100f33a29044fa9d740c9b3213a93e57c84b472c84e0b8a0e1ae48e67a9f8f6de9d5f7f3d52ac23e48ac51801f1dc950abe901da34d2a9e3baadb141a17c77ef3c565dd5ee5054b91cf63bb3c6ab83f72ab3aafe93d0fc3c2348b764fafb0b1c0733de51459aeab46580384bf9d74c4e28164b7cde247f891ba07891c9d872ad2bb")]
```
Comment (nguerrera) [inline src/Microsoft.DotNet.Cli.Sln.Internal/Properties/AssemblyInfo.cs]: I'm missing why this is the only one that changed. The change to the csproj for the key seems to be the same as the one's below.

File: test/Microsoft.DotNet.Tools.Tests.Utilities/RepoDirectoriesProvider.cs [hunk 3]
```diff
@@ -58,46 +57,35 @@ public static string RepoRoot
...
-            if (string.IsNullOrEmpty(_testPackages))
-            {
-                throw new InvalidOperationException("TEST_PACKAGES environment variable not set");
-            }
+            _testPackages = Path.Combine(_artifacts, "tmp", configuration, "test", "packages");
+            // _testPackages = Environment.GetEnvironmentVariable("TEST_PACKAGES");
+            // if (string.IsNullOrEmpty(_testPackages))
+            // {
+            //     throw new InvalidOperationException("TEST_PACKAGES environment variable not set");
+            // }
```
Comment (nguerrera) [inline test/Microsoft.DotNet.Tools.Tests.Utilities/RepoDirectoriesProvider.cs]: Remove commented out code.

Comment (nguerrera) [reply test/Microsoft.DotNet.Tools.Tests.Utilities/RepoDirectoriesProvider.cs]: This is fixed in a later commit. I'm giving up on reviewing commit by commit. It's overwhelming GitHub and CodeFlow.

File: test/Microsoft.DotNet.ToolPackage.Tests/SampleGlobalToolWithShim/consoledemo.csproj [hunk 4]
```diff
@@ -3,5 +3,9 @@
     <OutputType>Exe</OutputType>
     <TargetFramework>netcoreapp2.1</TargetFramework>
      <AssemblyName>consoledemo</AssemblyName>
+     <RestoreSources>
+      $(RestoreSources);
+      https://api.nuget.org/v3/index.json;
+     </RestoreSources>
```
Comment (nguerrera) [inline test/Microsoft.DotNet.ToolPackage.Tests/SampleGlobalToolWithShim/consoledemo.csproj]: Why is this needed?

File: test/dotnet-msbuild.Tests/GivenMsbuildForwardingApp.cs [hunk 5]
```diff
@@ -71,7 +72,7 @@ public void ItSetsOrIgnoresTelemetrySessionId()
             var envVar = "DOTNET_CLI_TELEMETRY_SESSIONID";
             var startInfo = new MSBuildForwardingApp(new string[0], msbuildPath)
                 .GetProcessStartInfo();
-            (startInfo.Environment[envVar] == null || int.TryParse(startInfo.Environment[envVar], out _))
+            (startInfo.Environment[envVar] == null || Guid.TryParse(startInfo.Environment[envVar], out _))
```
Comment (nguerrera) [inline test/dotnet-msbuild.Tests/GivenMsbuildForwardingApp.cs]: >Guid [](start = 54, length = 4)

How was this only caught by arcade move?

File: test/dotnet-msbuild.Tests/GivenDotnetStoreInvocation.cs [hunk 6]
```diff
@@ -18,6 +18,11 @@ public class GivenDotnetStoreInvocation
         private static readonly string WorkingDirectory = 
             TestPathUtilities.FormatAbsolutePath(nameof(GivenDotnetStoreInvocation));
 
+        public GivenDotnetStoreInvocation()
+        {
+            Telemetry.Telemetry.CurrentSessionId = null;
```
Comment (nguerrera) [inline test/dotnet-msbuild.Tests/GivenDotnetStoreInvocation.cs]: >Telemetry.Telemetry.CurrentSessionId = null; [](start = 12, length = 44)

Why?

File: test/dotnet-msbuild.Tests/GivenDotnetRestoreInvocation.cs [hunk 7]
```diff
@@ -18,6 +18,11 @@ public class GivenDotnetRestoreInvocation
         private static readonly string WorkingDirectory = 
             TestPathUtilities.FormatAbsolutePath(nameof(GivenDotnetRestoreInvocation));
 
+        public GivenDotnetRestoreInvocation()
+        {
+            Telemetry.Telemetry.CurrentSessionId = null;
```
Comment (nguerrera) [inline test/dotnet-msbuild.Tests/GivenDotnetRestoreInvocation.cs]: Hmm, this is sprinkled all over. Can we put it in one place with an explanation?

File: src/Microsoft.DotNet.Cli.Sln.Internal/Microsoft.DotNet.Cli.Sln.Internal.csproj [hunk 8]
```diff
@@ -2,13 +2,13 @@
 
   <PropertyGroup>
     <Description>SLN file reader/writer</Description>
-    <Version>$(SdkVersion)</Version>
     <TargetFramework>netstandard2.0</TargetFramework>
     <DebugType>portable</DebugType>
     <AssemblyName>Microsoft.DotNet.Cli.Sln.Internal</AssemblyName>
-    <AssemblyOriginatorKeyFile>../../tools/Key.snk</AssemblyOriginatorKeyFile>
+    <StrongNameKeyId>MicrosoftAspNet</StrongNameKeyId>
```
Comment (wli3) [inline src/Microsoft.DotNet.Cli.Sln.Internal/Microsoft.DotNet.Cli.Sln.Internal.csproj]: why internal csproj's key is MicrosoftAspNet?

File: test/dotnet-build.Tests/GivenDotnetBuildBuildsCsproj.cs [hunk 9]
```diff
@@ -22,6 +22,8 @@ public void ItBuildsARunnableOutput()
                 .WithSourceFiles()
                 .WithRestoreFiles();
 
+            Console.WriteLine($"PATH: {testInstance.Root}");
```
Comment (wli3) [inline test/dotnet-build.Tests/GivenDotnetBuildBuildsCsproj.cs]: nit: forget to remove?

File: build/AppHostTemplate.proj [hunk 10]
```diff
@@ -1,49 +0,0 @@
-<Project ToolsVersion="15.0" DefaultTargets="ExtractAppHostToOutput">
-  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
-  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.tasks))\dir.tasks" />
-
-  <PropertyGroup>
-    <NativeExecutableExtension Condition=" $(Rid.StartsWith('win'))">.exe</NativeExecutableExtension>
```
Comment (wli3) [inline build/AppHostTemplate.proj]: Want to make sure that this logic and its tests are in core-sdk now

File: src/dotnet/dotnet.csproj (hunk 1 above)
Comment (livarcocc) [reply src/dotnet/dotnet.csproj]: Yes.

File: src/Microsoft.DotNet.Cli.Sln.Internal/Properties/AssemblyInfo.cs (hunk 2 above)
Comment (livarcocc) [reply src/Microsoft.DotNet.Cli.Sln.Internal/Properties/AssemblyInfo.cs]: Internal.Tests was using some random key we had in the repo. I imagine that for a test project, it was fine to change the key. Which led to this needing to be updated as well.

File: test/Microsoft.DotNet.ToolPackage.Tests/SampleGlobalToolWithShim/consoledemo.csproj (hunk 4 above)
Comment (livarcocc) [reply test/Microsoft.DotNet.ToolPackage.Tests/SampleGlobalToolWithShim/consoledemo.csproj]: Because this project needs to be moved to test assets, but this change was already too big and I wanted to refrain from it. This feed is needed otherwise it fails to find some dependencies, rolls them forward and fails the test.

File: test/dotnet-msbuild.Tests/GivenMsbuildForwardingApp.cs (hunk 5 above)
Comment (livarcocc) [reply test/dotnet-msbuild.Tests/GivenMsbuildForwardingApp.cs]: Right. I saw it fail on a couple of runs in the past but never spent the time tracking it down. This time I did :Smile:

File: src/Microsoft.DotNet.Cli.Sln.Internal/Properties/AssemblyInfo.cs (hunk 2 above)
Comment (nguerrera) [reply src/Microsoft.DotNet.Cli.Sln.Internal/Properties/AssemblyInfo.cs]: OK, nevermind. First, I misread test_key as same as key, then I misread Sln.Internal as same as Sln.Internal.Tests.

File: test/dotnet-msbuild.Tests/GivenDotnetStoreInvocation.cs (hunk 6 above)
Comment (livarcocc) [reply test/dotnet-msbuild.Tests/GivenDotnetStoreInvocation.cs]: Exactly. Without this, some tests were failing randomly with the SessionId being set.

File: test/dotnet-build.Tests/GivenDotnetBuildBuildsCsproj.cs (hunk 9 above)
Comment (livarcocc) [reply test/dotnet-build.Tests/GivenDotnetBuildBuildsCsproj.cs]: Oops.

File: test/dotnet-msbuild.Tests/GivenDotnetStoreInvocation.cs (hunk 6 above)
Comment (nguerrera) [reply test/dotnet-msbuild.Tests/GivenDotnetStoreInvocation.cs]: OK, see other comment about how this is sprinkled around. Can we put it behind a method with an explanation.

**Comment by wli3** [comment]
i tried out, for CI build it looks good. but dev cycle need to adjust. **Please merge, i can send a PR later**. Script like `cli-build-env.bat` is still needed to open in vs and need update to reflect directory rename.

run-build.ps1 cannot pass `/t:preparetests` to get overlaid dotnet. And overlaid dotnet is at  `cli\artifacts\tmp\Debug\dotnet`?

Good new, VS test explore can run most of the tests now. But I am not sure if it is running the right one in tests assuming stage2.

File: eng/Versions.props [hunk 11]
```diff
@@ -14,31 +15,21 @@
 
   <!-- Production Dependencies -->
   <PropertyGroup>
-    <MicrosoftAspNetCoreAllPackageVersion>2.2.0-preview2-35157</MicrosoftAspNetCoreAllPackageVersion>
+    <DotNetCoreSdkLKGVersion>3.0.100-preview-009722</DotNetCoreSdkLKGVersion>
```
Comment (dsplaisted) [inline eng/Versions.props]: Is this duplicated with what's in global.json?  Should this read it from there instead?

File: build/test/OverlaySdkOnLKG.targets [hunk 12]
```diff
@@ -11,4 +12,8 @@
     <Copy SourceFiles="@(OverlayCLI)"
           DestinationFiles="@(OverlayCLI->'$(SdkOutputDirectory)\sdk\$(DotNetCoreSdkLKGVersion)\%(RecursiveDir)%(Filename)%(Extension)')" />
   </Target>
+
+  <Target Name="PublishRedist">
+    <Exec Command="$(DotNetExe) publish $(RepoRoot)/src/redist/redist.csproj --no-build --output $(CliOutputDirectory)" WorkingDirectory="$(RepoRoot)/src/redist" />
```
Comment (dsplaisted) [inline build/test/OverlaySdkOnLKG.targets]: Is there a reason to shell out to a new process for this?  That has the same downside as the `DotNetPublish` target would have (ie no logging).

File: build/test/OverlaySdkOnLKG.targets [hunk 13]
```diff
@@ -14,6 +14,6 @@
   </Target>
 
   <Target Name="PublishRedist">
-    <Exec Command="$(DotNetExe) publish $(RepoRoot)/src/redist/redist.csproj --no-build --output $(CliOutputDirectory)" WorkingDirectory="$(RepoRoot)/src/redist" />
+    <Exec Command="$(DotNetExe) publish $(RepoRoot)/src/redist/redist.csproj --output $(CliOutputDirectory)" WorkingDirectory="$(RepoRoot)/src/redist" />
```
Comment (dsplaisted) [inline build/test/OverlaySdkOnLKG.targets]: Change this to an `MSBuild` task invocation if possible

File: eng/restore-toolset.sh [hunk 14]
```diff
@@ -41,7 +41,7 @@ function InitializeCustomSDKToolset {
     return
   fi
 
-  InstallDotNetSharedFramework "1.1.2"
+  InstallDotNetSharedFramework "9.1.2"
```
Comment (dsplaisted) [inline eng/restore-toolset.sh]: 9.1.2?  That's got to be a typo.

File: eng/Versions.props (hunk 11 above)
Comment (livarcocc) [reply eng/Versions.props]: Yeah, we should take this from global.json. I will address this when working on enabling official builds based on arcade and dnceng, which I will start on next.

File: build/test/OverlaySdkOnLKG.targets (hunk 13 above)
Comment (livarcocc) [reply build/test/OverlaySdkOnLKG.targets]: Will do.

Comment (livarcocc) [reply build/test/OverlaySdkOnLKG.targets]: Actually, I did that already, but you are reviewing by commit it seems, so, in this commit this hadn't happened yet.

File: build/test/OverlaySdkOnLKG.targets (hunk 12 above)
Comment (livarcocc) [reply build/test/OverlaySdkOnLKG.targets]: This was already moved to msbuild in a subsequent commit.

File: eng/restore-toolset.sh (hunk 14 above)
Comment (livarcocc) [reply eng/restore-toolset.sh]: Already fixed in a subsequent commit. I was testing something when I accidentally commited this.

**Comment by livarcocc** [comment]
@dotnet/dotnet-cli I am going to go ahead and merge this. Our official builds will be down for a day or so while I enable dnceng builds.

File: eng/restore-toolset.sh [hunk 15]
```diff
@@ -0,0 +1,68 @@
+function InitializeCustomSDKToolset {
+  if [[ "$restore" != true ]]; then
+    return
+  fi
+
+  # Disable first run since we want to control all package sources
+  export DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1
```
[the remark this answers is not in the document]
Comment (livarcocc) [reply eng/restore-toolset.sh]: Done in a separate PR.

File: eng/restore-toolset.sh [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+
+  # Enable verbose VS Test Console logging
+  export VSTEST_BUILD_TRACE=1
+  export VSTEST_TRACE_BUILD=1
+
+
+  # Don't resolve shared frameworks from user or global locations
+  export DOTNET_MULTILEVEL_LOOKUP=0
```
[the remark this answers is not in the document]
Comment (livarcocc) [reply eng/restore-toolset.sh]: Done in a separate PR.

File: test/EndToEnd.Tests/EndToEnd.Tests.csproj [hunk 17]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
     <TargetFramework>$(CliTargetFramework)</TargetFramework>
-    <AssemblyName>EndToEnd</AssemblyName>
+    <AssemblyName>EndToEnd.Tests</AssemblyName>
```
[the remark this answers is not in the document]
Comment (livarcocc) [reply test/EndToEnd.Tests/EndToEnd.Tests.csproj]: Done in a separate PR.
